taichi/program/kernel.h

```diff
@@ -41,6 +41,13 @@ class Kernel : public Callable {
                                 uint64 size,
                                 bool is_device_allocation);
 
+    void set_arg_external_array_w_shape(int arg_id,
```

Comment:
nit: suggest to either use `with_` or just remove `w_` (because shape is already one of the params) :-)

Reply:
Thank you! I'll change the name in a moment :)

Reply:
I finally keep the `with_` as it reads better.